src/main/java/appeng/fluids/client/gui/GuiFluidLevelEmitter.java

```diff
@@ -0,0 +1,216 @@
+		this.buttonList.add( this.plus100 = new GuiButton( 0, this.guiLeft + 82, this.guiTop + 17, 32, 20, "+" + c ) );
+		this.buttonList.add( this.plus1000 = new GuiButton( 0, this.guiLeft + 120, this.guiTop + 17, 38, 20, "+" + d ) );
+
+		this.buttonList.add( this.minus1 = new GuiButton( 0, this.guiLeft + 20, this.guiTop + 59, 22, 20, "-" + a ) );
+		this.buttonList.add( this.minus10 = new GuiButton( 0, this.guiLeft + 48, this.guiTop + 59, 28, 20, "-" + b ) );
+		this.buttonList.add( this.minus100 = new GuiButton( 0, this.guiLeft + 82, this.guiTop + 59, 32, 20, "-" + c ) );
+		this.buttonList.add( this.minus1000 = new GuiButton( 0, this.guiLeft + 120, this.guiTop + 59, 38, 20, "-" + d ) );
+
+		this.buttonList.add( this.redstoneMode );
+	}
+
+	@Override
+	public void drawBG( final int offsetX, final int offsetY, final int mouseX, final int mouseY )
+	{
+		super.drawBG( offsetX, offsetY, mouseX, mouseY );
+		this.level.drawTextBox();
+	}
+
+	@Override
+	protected boolean drawUpgrades()
+	{
+		return false;
+	}
+
+	@Override
+	protected String getBackground()
+	{
+		return "guis/lvlemitter.png";
+	}
+
+	@Override
+	protected GuiText getName()
+	{
+		return GuiText.FluidLevelEmitter;
+	}
+
+	@Override
+	protected void handleButtonVisibility()
+	{
+	}
+
+	@Override
+	protected void actionPerformed( final GuiButton btn ) throws IOException
+	{
+		super.actionPerformed( btn );
+
+		final boolean isPlus = btn == this.plus1 || btn == this.plus10 || btn == this.plus100 || btn == this.plus1000;
+		final boolean isMinus = btn == this.minus1 || btn == this.minus10 || btn == this.minus100 || btn == this.minus1000;
+
+		if( isPlus || isMinus )
+		{
+			this.addQty( this.getQty( btn ) );
+		}
+	}
+
+	private void addQty( final long i )
+	{
+		try
+		{
+			String Out = this.level.getText();
+
+			boolean Fixed = false;
+			while( Out.startsWith( "0" ) && Out.length() > 1 )
+			{
+				Out = Out.substring( 1 );
+				Fixed = true;
+			}
+
+			if( Fixed )
+			{
+				this.level.setText( Out );
+			}
+
+			if( Out.isEmpty() )
+			{
+				Out = "0";
+			}
+
+			long result = Long.parseLong( Out );
+			result += i;
+			if( result < 0 )
+			{
+				result = 0;
+			}
+
+			this.level.setText( Out = Long.toString( result ) );
+
+			NetworkHandler.instance().sendToServer( new PacketValueConfig( "FluidLevelEmitter.Value", Out ) );
+		}
+		catch( final NumberFormatException e )
+		{
+			// nope..
+			this.level.setText( "0" );
+		}
+		catch( final IOException e )
+		{
+			AELog.debug( e );
+		}
+	}
+
+	@Override
+	protected void keyTyped( final char character, final int key ) throws IOException
+	{
+		if( !this.checkHotbarKeys( key ) )
+		{
+			if( ( key == 211 || key == 205 || key == 203 || key == 14 || Character.isDigit( character ) ) && this.level.textboxKeyTyped( character, key ) )
+			{
+				try
+				{
+					String Out = this.level.getText();
+
+					boolean Fixed = false;
+					while( Out.startsWith( "0" ) && Out.length() > 1 )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)